digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -165,14 +165,37 @@ pub async fn get_assets_by_owner(
     limit: u64,
     options: &Options,
 ) -> Result<Vec<FullAsset>, DbErr> {
-    let cond = Condition::all()
-        .add(asset::Column::Owner.eq(owner.clone()))
-        .add(asset::Column::Supply.gt(0));
+    let mut joins = Vec::new();
+
+    if options.show_fungible {
+        let rel = extensions::token_accounts::Relation::Asset
+            .def()
+            .rev()
+            .on_condition(move |left, right| {
+                Condition::all().add(
+                    Expr::tbl(right, token_accounts::Column::Mint)
+                        .eq(Expr::tbl(left, asset::Column::Id)),
+                )
+            });
+        joins.push(rel);
+    }
+
+    let cond = if options.show_fungible {
+        Condition::all().add(asset::Column::Supply.gt(0)).add(
+            Condition::any()
+                .add(asset::Column::Owner.eq(owner.clone()))
+                .add(token_accounts::Column::Owner.eq(owner.clone())),
+        )
+    } else {
+        Condition::all()
+            .add(asset::Column::Supply.gt(0))
+            .add(asset::Column::Owner.eq(owner.clone()))
+    };
 
     get_assets_by_condition(
         conn,
         cond,
-        vec![],
+        joins,
```

Comment:
```suggestion
        joins.unwrap_or_default(),
```

This becomes with the one conditiona.